src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -1,126 +1,188 @@
 package org.broadinstitute.dsde.agora.server.business
 
 import org.broadinstitute.dsde.agora.server.exceptions.{AgoraEntityNotFoundException, NamespaceAuthorizationException, ValidationException}
-import org.broadinstitute.dsde.agora.server.dataaccess.AgoraDao
+import org.broadinstitute.dsde.agora.server.dataaccess.{AgoraDao, ReadWriteAction}
 import org.broadinstitute.dsde.agora.server.dataaccess.permissions._
 import org.broadinstitute.dsde.agora.server.dataaccess.permissions.AgoraPermissions._
 import org.broadinstitute.dsde.agora.server.model.{AgoraApiJsonSupport, AgoraEntity, AgoraEntityProjection, AgoraEntityType}
+import slick.dbio.DBIO
 import spray.json._
 
-class AgoraBusiness {
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
 
-  def insert(agoraEntity: AgoraEntity, username: String): AgoraEntity = {
-    if (!NamespacePermissionsClient.getNamespacePermission(agoraEntity, username).canCreate &&
-        !NamespacePermissionsClient.getNamespacePermission(agoraEntity, "public").canCreate) {
-      throw new NamespaceAuthorizationException(AgoraPermissions(Create), agoraEntity, username)
+class AgoraBusiness(permissionsDataSource: PermissionsDataSource)(implicit ec: ExecutionContext) {
+
+  //Creates a fake extra permission with the desired permission level conditional on checkAdmin and the user being an admin
+  private def makeDummyAdminPermission(checkAdmin: Boolean, permClient: PermissionsClient, username: String, permLevel: AgoraPermissions) = {
```

Comment:
you should probably scroll past these `checkFoo` functions in your first readthrough and head down to the things that do things for more context